### PR TITLE
Better handle attributes in "huge" edge set mode

### LINKED ISSUE #97: Report error if edge attributes are requested in huge edge set mode

### PR DESCRIPTION
**Pull request by darabos**
Fixes #97, I hope. I couldn't figure out how to reproduce the issue. Best I can say is that this compiles and LynxKite still starts up after.

### CODE REVIEW

**Review by xandrew-lynx — APPROVED** [review]
Try with this one:

```
- id: Create-vertices_1
  inputs: {}
  operationId: Create vertices
  parameters:
    size: '400'
  parametricParameters: {}
  x: 364
  y: 205.703125
- id: Connect-vertices-on-attribute_1
  inputs:
    graph:
      boxId: Add-constant-vertex-attribute_1
      id: graph
  operationId: Connect vertices on attribute
  parameters: {}
  parametricParameters: {}
  x: 599.8921834858915
  y: 165.48401137684266
- id: Add-constant-vertex-attribute_1
  inputs:
    graph:
      boxId: Create-vertices_1
      id: graph
  operationId: Add constant vertex attribute
  parameters:
    name: c
  parametricParameters: {}
  x: 479.4664879420196
  y: 291.9445757911481
- id: Merge-vertices-by-attribute_1
  inputs:
    graph:
      boxId: Add-random-edge-attribute_1
      id: graph
  operationId: Merge vertices by attribute
  parameters: {}
  parametricParameters: {}
  x: 1047.903461835176
  y: 20.738557410251204
- id: Add-random-edge-attribute_1
  inputs:
    graph:
      boxId: Connect-vertices-on-attribute_1
      id: graph
  operationId: Add random edge attribute
  parameters:
    dist: Standard Uniform
  parametricParameters: {}
  x: 816.2948255124186
  y: 202.4532352452246
- id: Graph-visualization_1
  inputs:
    graph:
      boxId: Add-random-edge-attribute_1
      id: graph
  operationId: Graph visualization
  parameters:
    state: >-
      {"left":{"projectPath":"","graphMode":"sampled","display":"svg","filters":{"vertex":{},"edge":{"random":"<0.001"}},"bucketCount":4,"preciseBucketSizes":false,"relativeEdgeDensity":false,"axisOptions":{"vertex":{},"edge":{}},"sampleRadius":1,"attributeTitles":{"width":"random","edge
      label":"random"},"animate":{"enabled":false,"style":"expand","labelAttraction":0},"centers":["auto"],"customVisualizationFilters":false,"vertexColorMap":"Viridis","labelColorMap":"Viridis","edgeColorMap":"Viridis","sliderColorMap":"Blue
      to
      orange"},"right":{"display":"svg","filters":{"vertex":{},"edge":{}},"bucketCount":4,"preciseBucketSizes":false,"relativeEdgeDensity":false,"axisOptions":{"vertex":{},"edge":{}},"sampleRadius":1,"attributeTitles":{},"animate":{"enabled":false,"style":"expand","labelAttraction":0},"centers":["auto"],"customVisualizationFilters":false,"vertexColorMap":"Viridis","labelColorMap":"Viridis","edgeColorMap":"Viridis","sliderColorMap":"Blue
      to orange"}}
  parametricParameters: {}
  x: 1117.8894370392825
  y: 181.44184669198157
```

**Comment by darabos** [comment]
> Try with this one:

Thanks! Works fine.
image

File: app/com/lynxanalytics/biggraph/spark_util/RDDUtils.scala [hunk 1]
```diff
@@ -55,7 +56,7 @@ class IDBuckets[T](
   }
 }
 object IDBuckets {
-  val MaxSampleSize = 50
+  val MaxSampleSize = LoggedEnvironment.envOrElse("KITE_MAX_BUCKET_SAMPLE_SIZE", "1000").toInt
```
[the remark this answers is not in the document]
Comment (darabos) [reply app/com/lynxanalytics/biggraph/spark_util/RDDUtils.scala]: > What is this change for?

I thought this threshold was the trigger for "huge edge set". But it's actually `DrawingThresholds.SmallEdges` (at 50,000). I've now reverted this change.

**Comment by xandrew-lynx** [comment]
:+1:

**Pull request merged**